lib/client/src/client/client.ts

```diff
@@ -138,15 +189,15 @@ export interface Client<R extends Range> {
      * like RxJS), but it means that the client needs to manually poll for updated annotations if
      * freshness is important.
      */
-    annotations(params: AnnotationsParams): Promise<Annotation<R>[]>
+    annotations(params: AnnotationsParams): Promise<EachWithProviderURI<Annotation<R>[]>>
 
     /**
      * Observe OpenCtx annotations from the configured providers for the given resource.
      *
      * The returned observable streams annotations as they are received from the providers and
      * continues passing along any updates until unsubscribed.
      */
-    annotationsChanges(params: AnnotationsParams): Observable<Annotation<R>[]>
+    annotationsChanges(params: AnnotationsParams): Observable<EachWithProviderURI<Annotation<R>[]>>
```

Comment:
The slapping everywhere of EachWithProviderURI makes me wonder if we need to adjust the protocol rather to encode this information.

A type like this makes me sweat.

Reply:
What do you think is a better way to attach the provider URIs? I am open to ideas.

Reply:
Maybe each field just has provider as part of the schema, but documented that the client overwrite this field and it doesn't need to be set by the provider? But yeah maybe this is the best way to do it from a TS point of view. I am still developing my TS taste :)